types/yup/index.d.ts

```diff
@@ -26,6 +26,7 @@ export function addMethod<T extends Schema<any>>(
 export function ref(path: string, options?: { contextPrefix: string }): Ref;
 export function lazy<T>(fn: (value: T) => Schema<T>): Lazy;
 export function setLocale(customLocale: LocaleObject): void;
+export function isSchema(obj: any): boolean;
```

Comment:
My only thought here is if we can also use this as a typeguard in typescript as well as a runtime function because it feels like it should be able to serve both functions. I wonder if we can change the function signature to

```typescript
export function isSchema(obj: any): obj is Schema;
```

I haven't actually put this in the editor and run the tests against it or `tsc` so this syntax might not work but with the addition of `obj is Schema` we'll be able to use `isSchema` as a typeguard in typescript as well.

Reply:
You are right. Fixed.